### PR TITLE
Rewrite request URIs for pub client APIs.

### PR DESCRIPTION
**Pull request by isoos**
#2646

### CODE REVIEW

**Comment by isoos** [comment]
verified locally, integration tests are passing, publishing to staging

**Comment by isoos** [comment]
Staging did reveal one issue, I'm debugging what went wrong with it:
```
api_builder: Unhandled error in API handler (incidentId: 8aa1bcf9-67a2-4dd1-bc5c-8080c04fd6a0)

Error:
    Invalid argument(s): url "api/packages/versions/newUploadFinish?upload_id=4dd9a4fa-7661-4494-aeb4-ffd4da278557&bucket=dartlang-pub-dev--pub-packages&key=tmp/4dd9a4fa-7661-4494-aeb4-ffd4da278557&etag=8ab20641b37aaba45b33fea2ab4ea390" must have the same query parameters as requestedUri "https://dartlang-pub-dev.appspot.com/api/packages/versions/newUploadFinish?upload_id=4dd9a4fa-7661-4494-aeb4-ffd4da278557&bucket=dartlang-pub-dev--pub-packages&key=tmp%2F4dd9a4fa-7661-4494-aeb4-ffd4da278557&etag=8ab20641b37aaba45b33fea2ab4ea390".

Stack:
    #0      _computeUrl (package:shelf/src/request.dart:304:7)
    #1      new Request._ (package:shelf/src/request.dart:166:20)
    #2      new Request (package:shelf/src/request.dart:138:14)
    #3      _normalizeHost (package:pub_dartlang_org/frontend/handlers/pubapi.dart:264:10)
```

**Comment by isoos** [comment]
Oh, got it, no need to set `url` on the request. Updating here and on staging...

**Comment by isoos** [comment]
Upload looks good on staging.

File: app/lib/frontend/handlers/pubapi.dart [hunk 1]
```diff
@@ -252,3 +253,22 @@ class PubApi {
   Future<Response> packageJson(Request request, String package) =>
       packageShowHandlerJson(request, package);
 }
+
```
Comment (jonasfj) [inline app/lib/frontend/handlers/pubapi.dart]: What does this is do and why, this critically needs to be explained in a doc-comment :)

**Pull request merged**